src/nodes/constraint_aware_append/constraint_aware_append.c

```diff
@@ -64,8 +65,8 @@ get_plans_for_exclusion(Plan *plan)
 	{
 		case T_Result:
 		case T_Sort:
-			Assert(plan->lefttree != NULL && plan->righttree == NULL);
-			return plan->lefttree;
+			Ensure(plan->lefttree != NULL, "subplan is null");
```

Comment:
Shouldn't `Assert(plan->righttree == NULL)` still apply?

Reply:
plan->righttree will always be NULL for Sort or Result

Reply:
The difference between assert/ensure is the effect on the user when you are wrong about the assumption. assert are never run so in this specific case it would lead to a segfault while the ensure produces an internal error for the user running the query. I changed the assert into an ensure because while i dont think it can happen in a normal postgres instance we might end up here due to other bugs. We dont always have control over how the plan looks like as most of it is handled by postgres and additionally any extension can create further plans.